Sources/MapboxNavigation/NavigationMapView+VanishingRouteLine.swift

```diff
@@ -35,15 +35,20 @@ extension NavigationMapView {
      */
     func parseRoutePoints(route: Route) -> RoutePoints {
         let nestedList = route.legs.map { (routeLeg: RouteLeg) -> [[CLLocationCoordinate2D]] in
-            return routeLeg.steps.map { (routeStep: RouteStep) -> [CLLocationCoordinate2D] in
+            var legCoordinates = [[CLLocationCoordinate2D]]()
+            let endIndex = routeLeg.steps.count - 1
+            for (index, routeStep) in routeLeg.steps.enumerated() {
                 if let routeShape = routeStep.shape {
-                    if !routeShape.coordinates.isEmpty {
-                        return routeShape.coordinates
-                    } else { return [] }
+                    if index != endIndex {
+                        legCoordinates.append(routeShape.coordinates)
+                    } else if let finalStep = routeShape.coordinates.first {
+                        legCoordinates.append([finalStep])
```

Comment:
It would be clearer to map the steps to coordinates arrays, as before, but then drop the last coordinate if it’s a duplicate, or perhaps only do that for an arrival step.

Reply:
Thanks for the suggestion, I'll update this part.